File: src/2044_keep_multiplying_found_values_by_two/keep_multiplying_found_values_by_two.py
class Solution:
    def findFinalValue(self, nums, original):
        nums_set = set(nums)
        
        while original in nums_set:
            original *= 2

        return original

s = Solution()
nums = [5,3,6,1,12]
original = 3
print(s.findFinalValue(nums, original), 24)
nums = [2,7,9]
original = 4
print(s.findFinalValue(nums, original), 4)

File: src/2044_keep_multiplying_found_values_by_two/KeepMultiplyingFoundValuesByTwo.java
import java.util.HashSet;
import java.util.Set;

public class KeepMultiplyingFoundValuesByTwo {
    public int findFinalValue(int[] nums, int original) {
        Set<Integer> numsSet = new HashSet<>();

        for (int num : nums) {
            numsSet.add(num);
        }

        while (numsSet.contains(original))
            original *= 2;

        return original;
    }

    public static void main(String[] args) {
        KeepMultiplyingFoundValuesByTwo s = new KeepMultiplyingFoundValuesByTwo();
        int[] nums1 = { 5, 3, 6, 1, 12 };
        int original1 = 3;
        System.out.println(s.findFinalValue(nums1, original1) + " 24");
        int[] nums2 = { 2, 7, 9 };
        int original2 = 4;
        System.out.println(s.findFinalValue(nums2, original2) + " 4");
    }
}
